src/extra/others/fragment/lv_fragment.c

```diff
@@ -0,0 +1,392 @@
+    lv_fragment_manager_t * instance = lv_mem_alloc(sizeof(lv_fragment_manager_t));
+    instance->parent = parent;
+    instance->container = container;
+    instance->top = NULL;
+    return instance;
+}
+
+void lv_fragment_manager_del(lv_fragment_manager_t * manager)
+{
+    LV_ASSERT(manager);
+    manager_stack_t * top = manager->top;
+    while(top) {
+        LV_ASSERT(top->cls);
+        item_destroy_obj(manager, top);
+        item_destroy_fragment(top);
+        struct manager_stack_t * prev = top->prev;
+        lv_mem_free(top);
+        top = prev;
+    }
+    lv_mem_free(manager);
+}
+
+void lv_fragment_manager_push(lv_fragment_manager_t * manager, const lv_fragment_class_t * cls, void * args)
+{
+    LV_ASSERT(manager);
+    LV_ASSERT(cls);
+    manager_stack_t * item = item_new(cls);
+    lv_obj_t * parent = manager->container;
+    item_create_fragment(manager, item, args);
+    /* Destroy object of previous screen */
+    if(manager->top) {
+        item_destroy_obj(manager, manager->top);
+    }
+    item_create_obj(manager, item, parent, NULL);
+    manager_stack_t * top = manager->top;
+    item->prev = top;
+    manager->top = item;
+}
+
+void lv_fragment_manager_replace(lv_fragment_manager_t * manager, const lv_fragment_class_t * cls,
+                                 void * args)
+{
+    LV_ASSERT(manager);
+    LV_ASSERT(cls);
+    manager_stack_t * top = item_new(cls);
+    item_create_fragment(manager, top, args);
+    manager_stack_t * old = manager->top;
+    if(old) {
+        item_destroy_obj(manager, old);
+        item_destroy_fragment(old);
+        lv_mem_free(old);
+    }
+    manager->top = top;
+    item_create_obj(manager, top, manager->container, NULL);
+}
+
+size_t lv_fragment_manager_size(lv_fragment_manager_t * manager)
+{
+    size_t size = 0;
+    for(manager_stack_t * cur = manager->top; cur; cur = cur->prev) {
+        size++;
+    }
+    return size;
+}
+
+void lv_fragment_manager_pop(lv_fragment_manager_t * manager)
+{
+    LV_ASSERT(manager);
+    manager_stack_t * top = manager->top;
+    if(!top) return;
+    manager_stack_t * prev = top->prev;
+    bool dialog = top->dialog;
+    if(!dialog && prev) {
+        item_create_fragment(manager, prev, NULL);
+    }
+    item_destroy_obj(manager, top);
+    item_destroy_fragment(top);
+    lv_mem_free(top);
+    if(!dialog && prev) {
+        item_create_obj(manager, prev, manager->container, NULL);
+    }
+    manager->top = prev;
+}
+
+bool lv_fragment_manager_dispatch_event(lv_fragment_manager_t * manager, int which, void * data1, void * data2)
+{
+    LV_ASSERT(manager);
+    manager_stack_t * top = manager->top;
+    if(!top) return false;
+    lv_fragment_t * instance = top->instance;
+    if(!instance || !top->cls->event_cb) return false;
+    return top->cls->event_cb(instance, which, data1, data2);
+}
+
+#if LV_USE_MSGBOX
+
+void lv_fragment_manager_show(lv_fragment_manager_t * manager, const lv_fragment_class_t * cls, void * args)
```

Comment:
What is purpose of this?

Reply:
It's for showing msgbox. When a msgbox is added, it should remove existing view, so it has a separate stack.